warehouse/query-core/src/main/java/datawave/query/tables/keyword/KeywordQueryLogic.java

```diff
@@ -243,14 +267,38 @@ public void setupQuery(GenericQueryConfiguration genericConfig) throws Exception
                             config.getMaxContentChars(), config.getState().getPreferredViews(), config.getState().getLanguageMap());
             scanner.addScanIterator(cfg);
 
-            this.iterator = scanner.iterator();
+            // wrap the scanIterator in case there is nothing there and we have external content that needs to be transformed
+            final Iterator<Entry<Key,Value>> scanIterator = scanner.iterator();
+            this.iterator = scanIterator;
+            if (externalTagCloudPartitions != null) {
+                Iterator<Entry<Key,Value>> fieldedKeywordIterator = prepareFieldedIterator(externalTagCloudPartitions);
+                this.iterator = new IteratorChain<>(scanIterator, fieldedKeywordIterator);
+            }
+
```

Comment:
I have a recommendation on how to simplify this by and centralizing this code and the code from `prepareFieldedIterator` into the `setExtternalTagCloudPartitions` method. This way we don't need to have externalTagCloudPartition handling code dispersed throughout the class.

It's a bit hard to capture in the GitHub review UI though, so ask me when we're both online.

Reply:
latest commit reflects my cut at this